server/stores_test.go

```diff
@@ -0,0 +1,388 @@
+						return chronograf.Source{
+							ID:           1,
+							Name:         "my sweet name",
+							Organization: "0",
+							Role:         "editor",
+						}, nil
+					},
+				},
+			},
+			args: args{
+				organization: "0",
+				role:         "editor",
+			},
+			wants: wants{
+				source: chronograf.Source{
+					ID:           1,
+					Name:         "my sweet name",
+					Organization: "0",
+					Role:         "editor",
+				},
+			},
+		},
+		{
+			name: "Get editor source as admin",
+			fields: fields{
+				SourcesStore: &mocks.SourcesStore{
+					GetF: func(ctx context.Context, id int) (chronograf.Source, error) {
+						return chronograf.Source{
+							ID:           1,
+							Name:         "my sweet name",
+							Organization: "0",
+							Role:         "editor",
+						}, nil
+					},
+				},
+			},
+			args: args{
+				organization: "0",
+				role:         "admin",
+			},
+			wants: wants{
+				source: chronograf.Source{
+					ID:           1,
+					Name:         "my sweet name",
+					Organization: "0",
+					Role:         "editor",
+				},
+			},
+		},
+		{
+			name: "Get editor source as viewer",
+			fields: fields{
+				SourcesStore: &mocks.SourcesStore{
+					GetF: func(ctx context.Context, id int) (chronograf.Source, error) {
+						return chronograf.Source{
+							ID:           1,
+							Name:         "my sweet name",
+							Organization: "0",
+							Role:         "editor",
+						}, nil
+					},
+				},
+			},
+			args: args{
+				organization: "0",
+				role:         "viewer",
+			},
+			wants: wants{
+				err: true,
+			},
+		},
+		{
+			name: "Get admin source as admin",
+			fields: fields{
+				SourcesStore: &mocks.SourcesStore{
+					GetF: func(ctx context.Context, id int) (chronograf.Source, error) {
+						return chronograf.Source{
+							ID:           1,
+							Name:         "my sweet name",
+							Organization: "0",
+							Role:         "admin",
+						}, nil
+					},
+				},
+			},
+			args: args{
+				organization: "0",
+				role:         "admin",
+			},
+			wants: wants{
+				source: chronograf.Source{
+					ID:           1,
+					Name:         "my sweet name",
+					Organization: "0",
+					Role:         "admin",
+				},
+			},
+		},
+		{
+			name: "Get source as super admin",
+			fields: fields{
+				SourcesStore: &mocks.SourcesStore{
+					GetF: func(ctx context.Context, id int) (chronograf.Source, error) {
+						return chronograf.Source{
+							ID:           1,
+							Name:         "my sweet name",
+							Organization: "0",
+							Role:         "viewer",
+						}, nil
+					},
+				},
+			},
+			args: args{
+				organization: "0",
+				role:         "viewer",
+				superAdmin:   true,
+			},
+			wants: wants{
+				source: chronograf.Source{
+					ID:           1,
+					Name:         "my sweet name",
+					Organization: "0",
+					Role:         "viewer",
+				},
+			},
+		},
+		{
+			name: "Get source as super admin - no organization or role",
```

Comment:
this test and the two below look wrong to me. superadmins should be able to get this source, even if they have no organization or role.